packages/next/server/app-render.tsx

```diff
@@ -83,6 +83,17 @@ const INTERNAL_COOKIES_INSTANCE = Symbol('internal for cookies readonly')
 function readonlyCookiesError() {
   return new Error('ReadonlyCookies cannot be modified')
 }
+
+function lazy<T>(fn: () => T) {
+  let value: any
+  return () => {
+    if (!value) {
+      value = fn()
+    }
+    return value
+  }
+}
+
```

Comment:
```suggestion
```

Could be removed per https://github.com/vercel/next.js/pull/41353/files#r993344264